src/Microsoft.JSInterop/JavaScriptRuntime/src/Microsoft.JSInterop.ts

```diff
@@ -231,4 +231,48 @@ module DotNet {
       throw new Error(`The value '${resultIdentifier}' is not a function.`);
     }
   }
+
+  class DotNetObject {    
+    constructor(private _id: string) {
+    }
+
+    public invokeMethod<T>(methodIdentifier: string, ...args: any[]): T {
+      return invokeMethod<T>(
+        'Microsoft.JSInterop',
+        'DotNetDispatcher.InvokeInstanceMethod',
+        this._id,
+        methodIdentifier,
+        JSON.stringify(args));
+    }
+
+    public invokeMethodAsync<T>(methodIdentifier: string, ...args: any[]): Promise<T> {
+      return invokeMethodAsync<T>(
+        'Microsoft.JSInterop',
+        'DotNetDispatcher.InvokeInstanceMethodAsync',
+        this._id,
+        methodIdentifier,
+        JSON.stringify(args));
+    }
+
+    public dispose() {
+      const promise = invokeMethodAsync<any>(
+        'Microsoft.JSInterop',
+        'DotNetDispatcher.ReleaseDotNetObject',
+        this._id);
+      promise.catch(error => console.error(error));
```

Comment:
Should we have some kind of global sink for errors that js swallows? I'm sure ultimately they would all end up in the console, but I'm wondering if its worth centralizing?

Reply:
I'm not sure. A developer can already register their own top-level uncaught exception handler if they want to (`window.onerror`). If it turns out there are reasons for adding further APIs around this we can do so.

Reply:
Actually I think code here is wrong, and should do something like:

    promise.catch(error => throw error);

.... to make sure it does hit `window.onerror`. I might be misremembering the details so will have to actually try it out.